pkg/controller/hibernation/aws_actuator.go

```diff
@@ -45,87 +48,210 @@ func (a *awsActuator) CanHandle(cd *hivev1.ClusterDeployment) bool {
 }
 
 // StopMachines will stop machines belonging to the given ClusterDeployment
-func (a *awsActuator) StopMachines(cd *hivev1.ClusterDeployment, c client.Client, logger log.FieldLogger) error {
+func (a *awsActuator) StopMachines(cd *hivev1.ClusterDeployment, hiveClient client.Client, logger log.FieldLogger) error {
 	logger = logger.WithField("cloud", "aws")
-	awsClient, err := a.awsClientFn(cd, c, logger)
+	awsClient, err := a.awsClientFn(cd, hiveClient, logger)
 	if err != nil {
 		return err
 	}
-	instanceIDs, err := getClusterInstanceIDs(cd, awsClient, runningOrPendingStates, logger)
+
+	instances, err := getClusterInstanceIDs(cd, awsClient, runningOrPendingStates, logger)
 	if err != nil {
 		return err
 	}
-	if len(instanceIDs) == 0 {
+	if len(instances) == 0 {
 		logger.Warning("No instances were found to stop")
 		return nil
 	}
-	logger.WithField("instanceIDs", instanceIDs).Info("Stopping cluster instances")
-	_, err = awsClient.StopInstances(&ec2.StopInstancesInput{
-		InstanceIds: instanceIDs,
+	instances, spotInstances := filterOutSpotInstances(instances)
+	if err := a.stopOnDemandInstances(awsClient, instanceIDs(instances), logger); err != nil {
+		return err
+	}
+	if err := a.stopSpotInstances(awsClient, instanceIDs(spotInstances), logger); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *awsActuator) stopOnDemandInstances(awsClient awsclient.Client, instanceIDs []string, logger log.FieldLogger) error {
+	if len(instanceIDs) == 0 {
+		return nil
+	}
+
+	logger.WithField("instanceIDs", instanceIDs).Info("Stopping on-demand cluster instances")
+	_, err := awsClient.StopInstances(&ec2.StopInstancesInput{
+		InstanceIds: aws.StringSlice(instanceIDs),
 	})
 	if err != nil {
-		logger.WithError(err).Error("failed to stop instances")
+		logger.WithError(err).Error("failed to stop on-demand instances")
+		return err
 	}
-	return err
+	return nil
+}
+
+func (a *awsActuator) stopSpotInstances(awsClient awsclient.Client, instanceIDs []string, logger log.FieldLogger) error {
+	if len(instanceIDs) == 0 {
+		return nil
+	}
+	logger.WithField("instanceIDs", instanceIDs).Info("Terminating spot cluster instances")
+	_, err := awsClient.TerminateInstances(&ec2.TerminateInstancesInput{
```

Comment:
It would be nice to include a comment (here or at the top of the func) with some of the text from the commit message describing why this is `TerminateInstances` rather than `StopInstances`.

Reply:
or perhaps just call the function `terminateSpotInstnaces`